### DISCUSSION TITLE
save scd file using OSC

### DISCUSSION THREAD

**Discussion by sjomae** [davidgranstrom/scnvim#235] [General]
Saving the edited *.scd file is pure a functionality of nvim?

I'm wondering if it is possible to save a *scd file using OSC (open sound control), by sending a OSC message to supercollider or (sc)nvim.

**Comment by davidgranstrom** [comment]
Writing the buffer (.scd) to disk is indeed performed by nvim.

It is possible however to send lua code to nvim via the `SCNvim` SuperCollider class. Note that scnvim/SuperCollider must be running for this to work. For instance, add the following OSC handler by executing the code below:

```supercollider
(
OSCdef(\writeFile, {
    SCNvim.luaeval("vim.cmd('write')")
}, "/writeFile")
)
```

Now if you send an OSC message to SuperCollider with the `/writeFile` address, nvim will write the current (active) buffer to disk, just like issuing a `:write` command inside nvim.

If you're looking for a more general solution of controlling nvim via OSC (not necessarily involving SuperCollider) you could take a look at [osc.nvim](https://github.com/davidgranstrom/osc.nvim).

**Reply by sjomae** [reply]
Thanks, this is helpful.  I'll look into this.
